corehq/apps/reports/filters/case_list.py

```diff
@@ -74,3 +99,19 @@ def get_default_selections(self):
             return self.default_selections
         else:
             return self._get_assigned_locations_default()
+
+
+@location_safe
+class CaseListFilterAllUsers(CaseListFilter):
```

Comment:
I see the original logic being referenced in a lot of worker monitoring reports, which I thought was outside the scope of this change. I'm also not sure how the monitoring reports are supposed to look bc they seem to have a different format than the case list report, so let's offline about this?

Reply:
Where do you see that?  I see it being referenced as a filter only the once, and then I see a couple places that get the URL from the class to redirect it to the report, some references in exports (which I think _are_ in scope, is that right?).  

At any rate, I think the thing called `CaseListFilter` should be the version used in the Case List Report, and if we also need different versions, than those should be named after what they are (maybe `CaseExportFilter`, for instance).

Reply:
There is 1 functional difference: deactivated users show up as options in the dropdown list (which is happening in `CaseListFilterUtilsAllUsers` because it subclasses `SubmitHistoryUtils`, which has a `DEACTIVATED` option). The `static_options` are overridden because overwrite `Deactivated Users` would show up in the dropdown. 

I'm not seeing `CaseListFilterAllUsers` duplicating logic, but I do see `CaseListFilterUtilsAllUsers` has some, I can clean that up.